compiler/crates/relay-codegen/src/printer.rs

```diff
@@ -470,7 +470,15 @@ impl<'b> JSONPrinter<'b> {
             },
             Primitive::JSModuleDependency(key) => match self.js_module_format {
                 JsModuleFormat::CommonJS => {
-                    self.write_js_dependency(f, key.to_string(), format!("./{}", key))
+                    let mut path = format!("./{}", key);
+
+                    let extension = &path[path.len() - 3..path.len()];
+
+                    if extension == ".js" || extension == ".ts" {
+                        path = path[0..path.len() - 3].to_string();
+                    }
```

Comment:
> the Path/PathBuf types have utilities for extracting the extension, let's use those? there is also the camino crate, which is basically utf8-only versions of Path/PathBuf which might help since the string is already utf8.

-- @josephsavona (internal review)

Reply:
missing `tsx`

Reply:
fixed https://github.com/facebook/relay/pull/4303